tests/integration/test_logcollector/test_command_monitoring/test_command_execution_freq.py

```diff
@@ -0,0 +1,115 @@
+    {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo full_command frequency 300', 'FREQUENCY': 300},  # 5 minutes.
+    {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo full_command frequency 1800', 'FREQUENCY': 1800},  # 30 minutes.
+    {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo full_command frequency 3600', 'FREQUENCY': 3600},  # 1 hour.
+    {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo full_command frequency 86400', 'FREQUENCY': 86400}  # 24 hours.
+]
+metadata = [
+    {'log_format': 'command', 'command': 'echo execution frequency 300', 'frequency': 300},  # 5 minutes.
+    {'log_format': 'command', 'command': 'echo execution frequency 1800', 'frequency': 1800},  # 30 minutes.
+    {'log_format': 'command', 'command': 'echo execution frequency 3600', 'frequency': 3600},  # 1 hour.
+    {'log_format': 'command', 'command': 'echo execution frequency 86400', 'frequency': 86400},  # 24 hours.
+    {'log_format': 'full_command', 'command': 'echo full_command frequency 300', 'frequency': 300},  # 5 minutes.
+    {'log_format': 'full_command', 'command': 'echo full_command frequency 1800', 'frequency': 1800},  # 30 minutes.
+    {'log_format': 'full_command', 'command': 'echo full_command frequency 3600', 'frequency': 3600},  # 1 hour.
+    {'log_format': 'full_command', 'command': 'echo full_command frequency 86400', 'frequency': 86400}  # 24 hours.
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['LOG_FORMAT'], x['COMMAND'], x['FREQUENCY']}" for x in parameters]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get configurations from the module."""
+    return local_internal_options
+
+
+def test_command_execution_freq(get_local_internal_options, configure_local_internal_options, get_configuration,
+                                configure_environment, restart_logcollector):
+    """Check if the Wazuh run correctly with the specified command monitoring option "frequency".
+
+    For this purpose, it is verified that the command has not been executed
+    before the period established in this option.
+
+    Args:
+        get_local_internal_options (fixture): Get internal configuration.
+        configure_local_internal_options (fixture): Set internal configuration.
+        get_configuration (fixture): Get configurations from the module.
+        configure_environment (fixture): Configure a custom environment for testing.
+        restart_logcollector (fixture): Reset log file and start a new monitor.
+
+    Raises:
+        TimeoutError: If the command monitoring callback is not generated.
+        AssertError: In the case of a server instance, the API response is different that the real configuration.
```

Comment:
This AssertError does not correspond to this test.
```suggestion
        AssertError: In the case of a server instance, the API response is different that the real configuration.
```

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/1257/commits/1e33fb8c458aeb8a738e22706713ec98fbf25703